src/components/modal/lagrede-filter/lagrede-filter-meny.tsx

```diff
@@ -0,0 +1,79 @@
+import {Normaltekst} from "nav-frontend-typografi";
+import React from "react";
+import {Hovedknapp, Knapp} from "nav-frontend-knapper";
+import {Visningstype} from "./lagre-filter-modal";
+import {useSelector} from "react-redux";
+import {AppState} from "../../../reducer";
+
+const lagreNyttFilterKnapp = (setValgtVisningstype) => {
+    return (
+        <Hovedknapp className="ny-knapp blokk-xs"
+                    onClick={() => setValgtVisningstype(Visningstype.LAGRE_NYTT)}>
+            Lagre som nytt filter
+        </Hovedknapp>
+    )
+}
+
+const oppdaterFilterKnapp = (setValgtVisningstype) => {
+    return (
+        <Knapp className="eksisterende-knapp"
+               onClick={() => setValgtVisningstype(Visningstype.OPPDATER)}>
+            Oppdater eksisterende filter
+        </Knapp>
+    )
+}
+
+const navnEllerFnrErBruktInnhold = () => {
+    return (
+        <>
+            <Normaltekst>Fødselsnummer og navn kan ikke brukes i lagrede filter.</Normaltekst>
+            <Normaltekst>Du må fjerne fødselsnummer og navn for å lagre filteret.</Normaltekst>
+        </>
+    )
+}
+
+const oppdatereFilterInnhold = (filterNavn, setValgtVisningstype) => {
+    return (
+        <>
+            <Normaltekst>Det finnes allerede et lagret
+                filter <b>"{filterNavn}"</b> med
+                denne filterkombinasjonen. Oppdater navnet ved å klikke på knappen under.
+            </Normaltekst>
+            <br/>
+            {oppdaterFilterKnapp(setValgtVisningstype)}
+        </>
+    )
+}
+
+const oppdatereEllerLagreNyttFilterInnhold = (filterNavn, setValgtVisningstype) => {
+    return (
+        <>
+            {lagreNyttFilterKnapp(setValgtVisningstype)}
+            <Normaltekst>Oppdater <b>"{filterNavn}"</b> ved å klikke
+                på
+                knappen under.</Normaltekst>
+            {oppdaterFilterKnapp(setValgtVisningstype)}
+        </>
+    )
+}
+const lagreNyttFilterInnhold = (setValgtVisningstype) => {
+    return lagreNyttFilterKnapp(setValgtVisningstype)
+}
+
+export function Meny(props: { setValgtVisningstype, sisteFilterNavn, erNavnEllerFnrBrukt }) {
+    const {valgtLagretFilter, sisteValgteLagredeFilter} = useSelector((state: AppState) => state.lagretFilter)
+
+    const getInhold = () => {
```

Comment:
is it necessary that getInhold is a function?

Reply:
I tried to have if statements istead of function call, but it was not working for some reason...so I made function